### ISSUE TITLE
Uncaught Error: Cannot find module 'native-image'

### ISSUE THREAD

**Issue by lolusab** [markushedvall/plantuml-viewer#25]
[Enter steps to reproduce below:]

1. create a .puml file
2. open plantuml viewer (ctrl alt p)

**Atom Version**: 1.11.2
**Electron Version**: 1.4.5
**System**: linux 4.8.6-1-ARCH
**Thrown From**: [plantuml-viewer](https://github.com/markushedvall/plantuml-viewer) package, v0.7.0

### Stack Trace

Uncaught Error: Cannot find module 'native-image'

```
At module.js:457

Error: Cannot find module 'native-image'
    at Module._resolveFilename (module.js:455:15)
    at Module._resolveFilename (/usr/lib/electron/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._resolveFilename (/usr/lib/atom/src/module-cache.js:383:52)
    at Function.Module._load (module.js:403:25)
    at Object.<anonymous> (/home/loic/.atom/packages/plantuml-viewer/lib/plantuml-viewer-view.js:12:19)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Object.<anonymous> (/home/loic/.atom/packages/plantuml-viewer/lib/plantuml-viewer-editor.js:5:26)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at isMarkdownViewerView (/home/loic/.atom/packages/plantuml-viewer/lib/plantuml-viewer.js:96:28)
    at atom-workspace.toggle (/home/loic/.atom/packages/plantuml-viewer/lib/plantuml-viewer.js:48:7)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/lib/atom/src/command-registry.js:260:29)
    at /usr/lib/atom/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/lib/atom/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/lib/atom/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/lib/atom/src/window-event-handler.js:114:36)
    at HTMLDocument.<anonymous> (/usr/lib/atom/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:02.7.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:01.5.0 plantuml-viewer:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "telemetryConsent": "limited"
  }
}
```

### Installed Packages

```coffee
# User
plantuml-viewer, v0.7.0 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.2.0 (inactive)
base16-tomorrow-light-theme, v1.2.0 (inactive)
one-dark-ui, v1.6.0 (active)
one-light-ui, v1.6.0 (inactive)
one-dark-syntax, v1.3.0 (active)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.7.0 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.201.1 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.59.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v1.0.0 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2-hotfix1 (active)
snippets, v1.0.2 (active)
spell-check, v0.68.2 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.101.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.209.3 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by bapti** [comment]
I'm also seeing this issue. It's odd as it's only just started happening, everything was working this morning and then this afternoon with no changes it's not not working. I wonder if Atom has updated in the background and made the breaking change

**Comment by dobinator** [comment]
uncomment line 12 in plantuml-viewer-view.js ... since this library is only used for copying a diagram and not for rendering it within atom it's kind of a workaround to be able to view the diagram until the issue is solved

**Comment by lolusab** [comment]
If I comment line 12 this is okay. Thanks for the solution

**Comment by dobinator** [comment]
I wouldn't close the issue though, because the issue still remains. What I posted is just a workaround until the problem is solved

**Comment by markushedvall** [comment]
The native-image module is from Electron. They might have changed the API for it though since this error is happening.

I'll see if this is a quick fix, otherwise I'll disable this feature.

**Comment by markushedvall** [comment]
The newest Atom release (1.12.0) has upgraded Electron to 1.3 and the way to include nativeImage has changed from:
```js
var nativeImage = require('native-image')
```
to:
```js
var nativeImage = require('electron').nativeImage
```

I'll have a new release fixing this issue out soon.

**Comment by markushedvall** [comment]
Fixed in 0.7.1 release
